server/src/main/java/org/opensearch/index/store/RemoteDirectory.java

```diff
@@ -230,9 +233,9 @@ public void close() throws IOException {
     @Override
     public long fileLength(String name) throws IOException {
         // ToDo: Instead of calling remote store each time, keep a cache with segment metadata
-        Map<String, BlobMetadata> metadata = blobContainer.listBlobsByPrefix(name);
-        if (metadata.containsKey(name)) {
-            return metadata.get(name).length();
+        List<BlobMetadata> metadata = blobContainer.listBlobsByPrefixInSortedOrder(name, 1, BlobContainer.BlobNameSortOrder.LEXICOGRAPHIC);
+        if (metadata.size() == 1 && metadata.get(0).name().equals(name)) {
+            return metadata.get(0).length();
```

Comment:
why do we need this change ?

Reply:
This is just to make sure we don't iterate over the results. Though we supply `name`, we don't expect it anyway, but we did see the  stack trace in hot threads dump.